src/lib/ldap/sync.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+/*
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or (at
+ *   your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, write to the Free Software
+ *   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/**
+ * $Id$
+ * @file lib/ldap/sync.h
+ *
+ * @brief Common definitions required by both network and worker for LDAP sync
+ *
+ * @copyright 2022 Network RADIUS SARL (legal@networkradius.com)
+ */
+
+/** Types of the internal packets for processing LDAP sync messages
+ */
+typedef enum {
```

Comment:
These should be available automatically (not as an enum, tho) as `src/include/protocol/ldap/freeradius.internal.h`

there are some admittedly horrific Perl scripts which convert the dictionaries to header files.

Reply:
Yeah best solution is to use the attribute macros as the values then they're in sync but we still have an enum for debugging.